libsolidity/parsing/Parser.cpp

```diff
@@ -2072,21 +2074,40 @@ optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> cons
 			sequencesToSearch.emplace_back(source.begin() + node->location().end, source.end());
 		}
 
-	vector<string> matches;
+	vector<string> licenseNames;
 	for (auto const& [start, end]: sequencesToSearch)
 	{
-		smatch match;
-		if (regex_search(start, end, match, licenseRegex))
-		{
-			string license{boost::trim_copy(string(match[1]))};
-			if (!license.empty())
-				matches.emplace_back(std::move(license));
-		}
+		auto declarationsBegin = std::sregex_iterator(start, end, licenseDeclarationRegex);
+		auto declarationsEnd = std::sregex_iterator();
+
+		for (std::sregex_iterator declIt = declarationsBegin; declIt != declarationsEnd; ++declIt)
+			if (!declIt->empty())
+			{
+				string license = boost::trim_copy(string((*declIt)[1]));
+
+				// Remove trailing comment terminator if present
+				if (boost::algorithm::ends_with(license, "*/"))
+					license = license.substr(0, license.size() - 2);
+
+				license = boost::trim_copy(license);
+
+				licenseNames.emplace_back(std::move(license));
```

Comment:
```suggestion
				licenseNames.emplace_back(move(license));
```

Reply:
Hmm I thought we wanted ot use `std::` for move calls? Though, looking at the source, it seems wildly inconsistent, but with a clear majority winner:

```
> git grep -o -E "(std::)?move\("  | grep -c std
619
> git grep -o -E "(std::)?move\("  | grep -c std -v
348

```

Reply:
We did? Ok then. It's just that in this file `move` is mostly used without `std::` and the `std::` prefix is generally superfluous if you have `using namespace std` in the file.

Reply:
We had it at a point where `boost::move` was still widespread. It is not anymore.

Reply:
I am still unclear at this point if you are saying we should use it or not @chriseth :)

Reply:
No need to use it, but also does not hurt.